src/any_agent/tools/mcp/frameworks/llama_index.py

```diff
@@ -14,51 +17,76 @@
     mcp_available = True
 
 
-class LlamaIndexMCPServerBase(MCPServerBase, ABC):
-    client: LlamaIndexMCPClient | None = None
-    framework: Literal[AgentFramework.LLAMA_INDEX] = AgentFramework.LLAMA_INDEX
+class LlamaIndexMCPConnection(MCPConnection, ABC):
+    """Base class for LlamaIndex MCP connections."""
 
-    def _check_dependencies(self) -> None:
-        """Check if the required dependencies for the MCP server are available."""
-        self.libraries = "any-agent[mcp,llama_index]"
-        self.mcp_available = mcp_available
-        super()._check_dependencies()
+    _client: LlamaIndexMCPClient | None = PrivateAttr(default=None)
 
     @abstractmethod
```

Comment:
Maybe I missed this from a previous PR: this isn't an abstract method so `@abstractmethod` doesn't apply anymore, right?

Reply:
Replied in the PR main section